From aa41c27233b85feb94e95270f0a54ed608eb6223 Mon Sep 17 00:00:00 2001
From: Masami Ichikawa <masami256@gmail.com>
Date: Mon, 29 Sep 2025 23:06:26 +0900
Subject: [PATCH] Enable Linux IMA feature

Enable IMA in kernel config and enable it in the linux kernel command line.

Signed-off-by: Masami Ichikawa <masami256@gmail.com>
---
 kconfigs/qemu.conf           | 1 +
 kconfigs/u-boot_qemu_v8.conf | 2 +-
 qemu_v8.mk                   | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/kconfigs/qemu.conf b/kconfigs/qemu.conf
index b39a7cee0007..b09c0ffa59a7 100644
--- a/kconfigs/qemu.conf
+++ b/kconfigs/qemu.conf
@@ -18,3 +18,4 @@ CONFIG_TCG_FTPM_TEE=y
 CONFIG_DEBUG_INFO=y
 CONFIG_DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT=y
 CONFIG_DEBUG_INFO_REDUCED is not set
+CONFIG_IMA=y
diff --git a/kconfigs/u-boot_qemu_v8.conf b/kconfigs/u-boot_qemu_v8.conf
index 9cc5822d9ddc..22008b2b5ee1 100644
--- a/kconfigs/u-boot_qemu_v8.conf
+++ b/kconfigs/u-boot_qemu_v8.conf
@@ -1,4 +1,4 @@
 CONFIG_SYS_TEXT_BASE=0x60000000
-CONFIG_BOOTCOMMAND="setenv kernel_addr_r 0x42200000 && setenv ramdisk_addr_r 0x46000000 && load hostfs - ${kernel_addr_r} uImage && load hostfs - ${ramdisk_addr_r} rootfs.cpio.uboot &&  setenv bootargs console=ttyAMA0,115200 earlyprintk=serial,ttyAMA0,115200 nokaslr oops=panic panic_on_warn=1 panic_on_oops=1 root=/dev/ram && bootm ${kernel_addr_r} ${ramdisk_addr_r} ${fdt_addr}"
+CONFIG_BOOTCOMMAND="setenv kernel_addr_r 0x42200000 && setenv ramdisk_addr_r 0x46000000 && load hostfs - ${kernel_addr_r} uImage && load hostfs - ${ramdisk_addr_r} rootfs.cpio.uboot &&  setenv bootargs console=ttyAMA0,115200 earlyprintk=serial,ttyAMA0,115200 nokaslr oops=panic panic_on_warn=1 panic_on_oops=1 root=/dev/ram ima=on ima_policy=tcb&& bootm ${kernel_addr_r} ${ramdisk_addr_r} ${fdt_addr}"
 CONFIG_SEMIHOSTING=y
 CONFIG_BOOTDELAY=0
diff --git a/qemu_v8.mk b/qemu_v8.mk
index b3984a619168..6ac8abb47e6d 100644
--- a/qemu_v8.mk
+++ b/qemu_v8.mk
@@ -600,6 +600,7 @@ QEMU_BASE_ARGS += -bios bl1.bin
 QEMU_BASE_ARGS += -initrd rootfs.cpio.gz
 QEMU_BASE_ARGS += -kernel Image
 QEMU_BASE_ARGS += -append 'console=ttyAMA0,38400 keep_bootcon nokaslr oops=panic panic_on_warn=1 panic_on_oops=1 root=/dev/vda2 $(QEMU_KERNEL_BOOTARGS)'
+QEMU_BASE_ARGS += -append 'console=ttyAMA0,38400 keep_bootcon nokaslr oops=panic panic_on_warn=1 panic_on_oops=1 root=/dev/vda2 $(QEMU_KERNEL_BOOTARGS) ima=on ima_policy=tcb'
 QEMU_BASE_ARGS += $(QEMU_XEN)
 QEMU_BASE_ARGS += $(QEMU_EXTRA_ARGS)
 QEMU_BASE_ARGS += -machine virt,acpi=off,secure=on,mte=$(QEMU_MTE),gic-version=$(QEMU_GIC_VERSION),virtualization=$(QEMU_VIRT)
-- 
2.51.0

